src/services/refactors/convertArrowFunctionOrFunctionExpression.ts

```diff
@@ -0,0 +1,192 @@
+        const statement = variableDeclarationList.parent;
+        if (!isVariableDeclarationList(variableDeclarationList) || !isVariableStatement(statement) || !isIdentifier(variableDeclaration.name)) return undefined;
+
+        return { variableDeclaration, variableDeclarationList, statement, name: variableDeclaration.name };
+    }
+
+    function getEditInfoForConvertToAnonymousFunction(context: RefactorContext, func: FunctionExpression | ArrowFunction): RefactorEditInfo {
+        const { file } = context;
+        const body = convertToBlock(func.body);
+        const newNode = createFunctionExpression(func.modifiers, func.asteriskToken, /* name */ undefined, func.typeParameters, func.parameters, func.type, body);
+        const edits = textChanges.ChangeTracker.with(context, t => t.replaceNode(file, func, newNode));
+        return { renameFilename: undefined, renameLocation: undefined, edits };
+    }
+
+    function getEditInfoForConvertToNamedFunction(context: RefactorContext, func: FunctionExpression | ArrowFunction): RefactorEditInfo | undefined {
+        const { file } = context;
+        const body = convertToBlock(func.body);
+        const variableInfo = getVariableInfo(func);
+        if (!variableInfo) return undefined;
+
+        const { variableDeclaration, variableDeclarationList, statement, name } = variableInfo;
+        const newNode = createFunctionDeclaration(func.decorators, statement.modifiers, func.asteriskToken, name, func.typeParameters, func.parameters, func.type, body);
+        let edits: FileTextChanges[];
+
+        if (variableDeclarationList.declarations.length === 1) {
+            edits = textChanges.ChangeTracker.with(context, t => t.replaceNode(file, statement, newNode));
+        }
+        else {
+            edits = textChanges.ChangeTracker.with(context, t => {
+                t.delete(file, variableDeclaration);
+                t.insertNodeAfter(file, statement, newNode);
+            });
+        }
+        return { renameFilename: undefined, renameLocation: undefined, edits };
+    }
+
+    function getEditInfoForConvertToArrowFunction(context: RefactorContext, func: FunctionExpression | ArrowFunction): RefactorEditInfo | undefined {
+        const { file } = context;
+        if (!isFunctionExpression(func)) return undefined;
+
+        const statements = func.body.statements;
+        const head = statements[0];
+        let body: ConciseBody;
+
+        if (canBeConvertedToExpression(func.body, head)) {
+            body = head.expression!;
+            suppressLeadingAndTrailingTrivia(body);
+            copyComments(head, body, file, SyntaxKind.MultiLineCommentTrivia, /* hasTrailingNewLine */ false);
+        }
+        else {
+            body = func.body;
+        }
+
+        const newNode = createArrowFunction(func.modifiers, func.typeParameters, func.parameters, func.type, /* equalsGreaterThanToken */ undefined, body);

```

Comment:
Suprised this works without the token -- seems like it would be better to `createToken(SyntaxKind.EqualsGreaterThanToken)` even if the emitter doesn't currently require it.